src/client.js

```diff
@@ -4059,15 +4060,38 @@ MatrixClient.prototype.doesServerSupportLazyLoading = async function() {
                 prefix: '',
             },
         );
+    }
+    return this._serverVersionsCache;
```

Comment:
we should expire this after ~8 hours, similar to how capabilities are dealt with. Concern would be someone who runs Riot for days on end (me) and upgrades their server in the background.

Reply:
Done: https://github.com/matrix-org/matrix-js-sdk/issues/1020